File: IsometricShooter/Scripts/JS/Effects/OnCollisionLegacyParticleEffect.js
#pragma strict

var legacyParticleEmitter : ParticleEmitter;

function OnCollisionEnter(collision : Collision) {

	if(legacyParticleEmitter != null) {
		for (var contact : ContactPoint in collision.contacts) {
			if(contact.otherCollider.gameObject.layer == 9) { //9 = Ground
				legacyParticleEmitter.transform.position = contact.point;
				legacyParticleEmitter.transform.position.y += 0.1; //correction to avoid flickering
				legacyParticleEmitter.Emit();
			}
		}
	}

}

File: IsometricShooter/Scripts/CS/Effects/OnCollisionLegacyParticleEffect.cs
using UnityEngine;
using System.Collections;

public class OnCollisionLegacyParticleEffect : MonoBehaviour {

	public ParticleEmitter legacyParticleEmitter;

	public void OnCollisionEnter(Collision collision) {

		if(legacyParticleEmitter != null) {
			foreach (ContactPoint contact in collision.contacts) {
				if(contact.otherCollider.gameObject.layer == 9) { //9 = Ground
					legacyParticleEmitter.transform.position = contact.point;
					float newY = legacyParticleEmitter.transform.position.y + 0.1f; //correction to avoid flickering
					legacyParticleEmitter.transform.position = new Vector3(legacyParticleEmitter.transform.position.x, newY, legacyParticleEmitter.transform.position.z);
					legacyParticleEmitter.Emit();
				}
			}
		}

	}
	
}
